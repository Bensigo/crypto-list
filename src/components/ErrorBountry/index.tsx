import React, { Component, Fragment } from 'react'
import Grid from "@material-ui/core/Grid"
import Typography from "@material-ui/core/Typography"
// import {withRouter, RouteComponentProps } from "react-router-dom"
import {connect } from "react-redux"
import { IState } from '../../reducers';


interface IApp {
  hasError: boolean
}
interface IProps   {
  error: string | null,

}

class ErrorBoundry extends Component <IProps,IApp>{
  constructor(props: any){
    super(props)
    this.state = {
      hasError: false
    }
  }
  componentDidCatch(info: any, error: any){
    this.setState({hasError: true})
    console.log(info)
    console.log(error)
  }
   static getDerivedStateFromError(error: any){
    return {
      hasError: true
    }
   }
   componentDidUpdate(prevProps:IProps, prevState: IApp){
    if (prevProps.error !== this.props.error){
      this.setState({hasError: true})
    }
   }
   
  render() {
    if (this.state.hasError){
      return (
      <Fragment>
          <Grid container justify="center" alignItems="center" style={{height: '70vh'}} >
          <Typography>{this.props.error}</Typography>
        </Grid>
      </Fragment>
      )
    }
    return this.props.children;
    
  }
}

const mapStateToProp = (state:IState) => ({
  error: state.shared.error
})

export default connect(mapStateToProp)(ErrorBoundry)


