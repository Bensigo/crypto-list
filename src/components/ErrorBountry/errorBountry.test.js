import React from 'react';
import ErrorBountry from "./index"
import {  configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Provider} from "react-redux"
import {BrowserRouter as Router} from "react-router-dom"
import Skeletonloading from "../CryptoListSekleton"
import configureStore from "redux-mock-store"

configure({adapter: new Adapter()});
describe("<ErrrorBountry/>", () => {
    it("check if snapshot match and error equal text",() => {
        const mockStore = configureStore()
        const store = mockStore({
            crypto: {
                coinMeta: null,
                coin: null,
                cryptos: []
            },
            shared: {
                loading: false,
                error: "Network error"
            }
        })
        // set error in state 
        const component = mount( 
            <Provider  store={store}>
                <Router>
                <React.Suspense fallback={<Skeletonloading/>}>
                    <ErrorBountry>
                       <h1>hello</h1>
                    </ErrorBountry>
                </React.Suspense>
                </Router>
            </Provider>       
            
        )
        expect(component).toMatchSnapshot()
        expect(component.props().store.getState().shared.error)
            .toEqual("Network error")
    } )
})