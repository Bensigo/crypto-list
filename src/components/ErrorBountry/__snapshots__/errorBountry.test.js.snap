// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrrorBountry/> check if snapshot match and error equal text 1`] = `ReactWrapper {}`;
